src/bus.rs

```diff
@@ -33,27 +47,50 @@ where
         self.pwr.set_high().unwrap();
         Timer::after(Duration::from_millis(250)).await;
 
-        while self.read32_swapped(REG_BUS_TEST_RO).await != FEEDBEAD {}
+        while self
+            .read32_swapped(REG_BUS_TEST_RO)
+            .inspect(|v| defmt::trace!("{:#x}", v))
+            .await
+            != FEEDBEAD
+        {}
 
         self.write32_swapped(REG_BUS_TEST_RW, TEST_PATTERN).await;
-        let val = self.read32_swapped(REG_BUS_TEST_RW).await;
+        let val = self
```

Comment:
instead of using `inspect`, just do a `trace!` write after the await. It's much straightforward.

Reply:
I removed all inspects()s except the while, althought that could be rewritten with loop {}, too.